src/controllers/EvidenceUploadController.ts

```diff
@@ -24,36 +32,100 @@ const navigation = {
     }
 };
 
+const continueButtonValidator: Validator = {
+
+    validate(request: Request): ValidationResult {
+
+        const appeal: Appeal = request.session
+            .chain(_ => _.getExtraData())
+            .map<ApplicationData>(data => data[APPLICATION_DATA_KEY])
+            .map(data => data.appeal)
+            .unsafeCoerce();
+
+        const attachments: Attachment[] | undefined = appeal.reasons.other.attachments;
+
+        if (!attachments || attachments.length === 0) {
+            return new ValidationResult([new ValidationError('file',
+                'You must add a document or click “Continue without adding documents”')]);
+        }
+        return new ValidationResult([]);
+    }
+};
+
 @controller(EVIDENCE_UPLOAD_PAGE_URI, SessionMiddleware, AuthMiddleware, FileTransferFeatureMiddleware)
 export class EvidenceUploadController extends BaseController<OtherReason> {
     constructor(@inject(FileTransferService) private readonly fileTransferService: FileTransferService) {
-        super(template, navigation);
+        super(template, navigation, continueButtonValidator);
     }
 
     protected prepareViewModelFromAppeal(appeal: Appeal): Record<string, any> & OtherReason {
         return appeal.reasons?.other;
     }
 
+    private async renderUploadError(appeal: Appeal, text: string): Promise<void> {
+        const validationResult: ValidationResult = new ValidationResult([
+            new ValidationError('file', text)
+        ]);
+
+        return await this.renderWithStatus(UNPROCESSABLE_ENTITY)(
+            this.template, {
+                ...this.prepareViewModelFromAppeal(appeal),
```

Comment:
Is "continue without docs" link working after validation error is triggered?

Reply:
Yes